### PR TITLE
Add new Dockerfile which can run test suite for PRs

### PR DESCRIPTION
**Pull request by matusmarhefka**
The new Dockerfile might be useful for debugging purposes or just for
running the test suite isolated from a host system. It runs the test suite
exactly the same as in GH action and it can be parametrized to run
for a specific GitHub PR.

### CODE REVIEW

File: Dockerfiles/test_suite [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#     podman start pr_test
+#     podman exec -it pr_test bash
+# and when you are done with the container you can remove it:
+#     podman rm -f pr_test
+#
+FROM fedora
+
+RUN dnf install -y git-extras git cmake dbus-devel GConf2-devel libacl-devel libblkid-devel libcap-devel libcurl-devel libgcrypt-devel libselinux-devel libxml2-devel libxslt-devel libattr-devel make openldap-devel pcre-devel perl-XML-Parser perl-XML-XPath perl-devel python3-devel rpm-devel swig bzip2-devel gcc-c++ libyaml-devel xmlsec1-devel xmlsec1-openssl-devel hostname bzip2 lua rpm-build which strace && dnf clean all
+
+RUN git clone https://github.com/OpenSCAP/openscap
```
Comment (jan-cerny) [inline Dockerfiles/test_suite:23]: Add `--recurse-submodules` to build and test also the yamlfilecontent probe.

```suggestion
RUN git clone --recurse-submodules https://github.com/OpenSCAP/openscap
```

**Review by jan-cerny — CHANGES_REQUESTED** [review]

**Comment by jan-cerny** [comment]
@matusmarhefka 
I have tried to use it to test PR #1781 and I can see that there is some fail:

```
183/258 Test #183: probes/file/test_probes_file_multiple_file_paths.sh ................................***Failed    0.07 sec
/bin/strace
/bin/strace: test_ptrace_get_syscall_info: PTRACE_TRACEME: Operation not permitted
/bin/strace: ptrace(PTRACE_TRACEME, ...): Operation not permitted
/bin/strace: PTRACE_SETOPTIONS: Operation not permitted
/bin/strace: detach: waitpid(38891): No child processes

```

Do I need to run podman as a `root` user?

**Comment by matusmarhefka** [comment]
> @matusmarhefka
> I have tried to use it to test PR #1781 and I can see that there is some fail:
> 
> ```
> 183/258 Test #183: probes/file/test_probes_file_multiple_file_paths.sh ................................***Failed    0.07 sec
> /bin/strace
> /bin/strace: test_ptrace_get_syscall_info: PTRACE_TRACEME: Operation not permitted
> /bin/strace: ptrace(PTRACE_TRACEME, ...): Operation not permitted
> /bin/strace: PTRACE_SETOPTIONS: Operation not permitted
> /bin/strace: detach: waitpid(38891): No child processes
> ```
> 
> Do I need to run podman as a `root` user?

Hmm, looks like a missing capability? Try `podman run --cap-add=SYS_PTRACE ...`.

**Comment by jan-cerny** [comment]
> Do I need to run podman as a root user?

You probably don't have to because the same problem occurs also when you run it as a root. It can be some missing permission such as capability?

**Comment by jan-cerny** [comment]
> Hmm, looks like a missing capability? Try podman run --cap-add=SYS_PTRACE ....

Yes! That was it! It fixed the problem for me.

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**